src/main/java/com/lambdatest/tunnel/Tunnel.java

```diff
@@ -584,58 +553,56 @@ public void runCommand(String command) throws IOException {
+                .setConnectTimeout(HTTP_TIMEOUT)
+                .setSocketTimeout(HTTP_TIMEOUT)
+                .setConnectionRequestTimeout(HTTP_TIMEOUT)
+                .build();
+        CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
 
-            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
-            String line = null;
+        for (int retryCount = 0; retryCount < MAX_TUNNEL_STARTUP_RETRY_COUNT; retryCount++) {
             try {
-                while ((line = reader.readLine()) != null) {
-                    if (line.contains("Downloading update")) {
-                        update = true;
-                    } else {
-                        if (update) {
-                            System.out.println("Tunnel is updated. restarting...");
-                            runCommandV2(command);
-                        }
-                        try {
-                            if (t1.port != null) {
-                                BufferedReader br = new BufferedReader(
-                                        new FileReader(String.valueOf(".lambdatest/tunnelprocs/" + t1.port) + ".txt"));
-                                if (br.readLine() != null) {
-                                    tunnelFlag = true;
-                                    System.out.println("Tunnel Started Successfully");
-                                    break;
-                                }
-                            }
-                        } catch (Exception e) {
-                            // System.out.println("Not found any file");
+                Thread.sleep(TUNNEL_INITIAL_WAIT_AND_RETRY_BACKOFF);
+                System.out.println("Checking Tunnel Status");
+                String infoAPIGetEndpoint = "http://127.0.0.1:" + Integer.toString(infoAPIPortValue) + TUNNEL_INFO_API;
+                HttpGet httpGet = new HttpGet(infoAPIGetEndpoint);
+                CloseableHttpResponse execute = httpClient.execute(httpGet);
+                if (execute.getStatusLine().getStatusCode() == 200) {
+                    System.out.println("Received Tunnel Status Response");
+                    BufferedReader reader = new BufferedReader(new InputStreamReader(execute.getEntity().getContent()));
+                    StringBuilder response = new StringBuilder();
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        response.append(line);
+                    }
+                    JSONObject tunnelInfo = new JSONObject(response.toString());
+                    if (tunnelInfo.getString("status").equals("SUCCESS")) {
+                        tunnelFlag = true;
+                        System.out.println("Tunnel Started Successfully");
+                        if (tunnelInfo.getJSONObject("data") != null && tunnelInfo.getJSONObject("data").has("id")) {

```

Comment:
should we go ahead and break even if the JSON object does not have the data key ?

Reply:
No we need this id. It should be present